### DISCUSSION TITLE
Improving documentation for Calibration

### DISCUSSION THREAD

**Discussion by Vinc0110** [scikit-rf/scikit-rf#644] [Ideas]
This is a fork of issue #590 intended to collect and discuss possible improvements of the API documentation, the tutorial, and the examples for the `Calibration` module. See https://scikit-rf.readthedocs.io/en/latest/api/calibration/index.html

Here is what I have or propose so far:
- Parameter `ideals` causes confusion and should be explained and documented in more detail in the API docs.
- Unifying the parameter description of the classes `Calibration` and its successors such that they share the same (verbose) docstring and that they are explained in the class docstring rather than in the docstring for `__init__()`. 
- Proposal: Entirely remove `__init__()` from the API docs and include parameter descriptions directly in the class docstrings **in every skrf module**, just like they do in NumPy. [The NumPy style guide](https://numpydoc.readthedocs.io/en/latest/format.html#documenting-classes) is not very explicit here, but see [numpy.ndarray](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html) as an example.
- Examples and use cases of calibrations with non-ideal (real-world) standards should be given [examples/tutorials]
- Add your points below

**Comment by Vinc0110** [comment]
- Improve parameter description of `TwoPortOnePath.apply_cal()`, see issue #623

**Comment by jhillairet** [comment]
* improve SOLT:  #624 and #627
